kernel/libs/aster-bigtcp/src/socket/bound.rs

```diff
@@ -451,6 +463,27 @@ impl<E> BoundTcpSocketInner<E> {
             return TcpProcessResult::NotProcessed;
         }
 
+        // If the socket is in the TimeWait state and a new packet arrives that is a SYN packet
+        // without ack number, the TimeWait socket will be marked as dead,
+        // and the packet will be passed on to any other listening sockets for processing.
+        //
+        // FIXME: Directly marking the TimeWait socket dead is not the correct approach.
+        // In Linux, a TimeWait socket remains alive to handle "old duplicate segments".
+        // If a TimeWait socket receives a new SYN packet, Linux will select a suitable
+        // listening socket from the socket table to respond to that SYN request.
+        // (https://elixir.bootlin.com/linux/v6.0.9/source/net/ipv4/tcp_ipv4.c#L2137)
+        // Moreover, the Initial Sequence Number (ISN) of the TimeWait socket will be set
+        // to prevent a timeout socket from erroneously handling packets from the new connection.
```

Comment:
This should be:
> Moreover, the Initial Sequence Number (ISN) **~~of the TimeWait socket~~** will be set to prevent **the TimeWait socket** from erroneously handling packets from the new connection.

 - There is no ISN for the TimeWait socket. It is for the new socket instead.
 - `a timeout socket` -> `the TimeWait socket`